titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/v1/V1SpecPodFactory.java

```diff
@@ -272,6 +281,40 @@ private V1Container buildV1ExtraContainers(BasicContainer extraContainer) {
                 .imagePullPolicy(DEFAULT_IMAGE_PULL_POLICY);
     }
 
+
+    private List<V1Volume> buildV1Volumes(List<Volume> volumes) {
+        if (volumes == null) {
+            return Collections.emptyList();
+        }
+        List<V1Volume> v1Volumes = new ArrayList<>();
+        for (Volume v : volumes) {
+            v1Volumes.add(buildV1Volume(v));
+        }
+        return v1Volumes;
+    }
+
+    private V1Volume buildV1Volume(Volume volume) {
+        if (volume.getVolumeSource() instanceof SharedContainerVolumeSource) {
+            V1FlexVolumeSource flexVolume = getV1FlexVolumeForSharedContainerVolumeSource(volume);
+            return new V1Volume()
+                    .name(volume.getName())
+                    .flexVolume(flexVolume);
+        } else {
+            // SharedVolumeSource is currently the only supported volume type
+            return null;
```

Comment:
I think this should never be null. Either we have Volume with valid data or the Volume list should be empty.